builds/misc/templates/build-packages.yaml

```diff
@@ -138,4 +138,220 @@ stages:
+      - job: mariner_linux_arm64
+      ################################################################################
+        displayName: Mariner_Linux on ARM64
+        pool:
+          name: $(pool.linux.arm.name)
+          demands:
+          - ImageOverride -equals agent-aziotedge-ubuntu-18.04-arm64
+        variables:
+          NugetSecurityAnalysisWarningLevel: warn
+        strategy:
+          matrix:
+            CBL-Mariner1.0-aarch64:
+              arch: aarch64
+              os: mariner
+              os_version: 1
+              mariner_release: 1.0-stable
+              target.iotedged: builds/mariner1/out/RPMS
+              target.logs: builds/mariner1/build/logs/pkggen/rpmbuilding/
+            CBL-Mariner2.0-aarch64:
+              arch: aarch64
+              os: mariner
+              os_version: 2
+              mariner_release: 2.0-stable
+              target.iotedged: builds/mariner2/out/RPMS
+              target.logs: builds/mariner2/build/logs/pkggen/rpmbuilding/
+
+        steps:
+          - bash: |
+              BASE_VERSION=`cat $BUILD_SOURCESDIRECTORY/edgelet/version.txt`
+              VERSION="$BASE_VERSION$BUILD_BUILDNUMBER"
+              echo "##vso[task.setvariable variable=VERSION;]$VERSION"
+              echo "##vso[task.setvariable variable=PACKAGE_ARCH;]$(arch)"
+              echo "PACKAGE_OS=$(os)"
+              echo "##vso[task.setvariable variable=MARINER_RELEASE;]$(mariner_release)"
+              echo "##vso[task.setvariable variable=OSVERSION;]$(os_version)"
+              mariner_arch=$(arch)
+              if [ $mariner_arch == "amd64" ]; then
+                mariner_arch="x86_64"
+              fi
+              echo "##vso[task.setvariable variable=MARINER_ARCH;]$mariner_arch"
+            displayName: Set Version
+
+          # remove this one 1ES can prevent the daily update from triggering
+          - bash: |
+              set -ex
+              ps aux | grep -i apt
+              waitAttempts=120
+              sleepDuration=10
+              for (( i=0; $i<$waitAttempts; i++ ))
+              do
+                if [[ $(ps aux | grep -c apt.systemd.daily) == 1 ]]; then
+                  break
+                fi
+                sleep $sleepDuration
+              done
+              ps aux | grep -i apt
+            displayName: Wait for 1ES agent update
+
+          - bash: |
+              sudo apt install apt-transport-https ca-certificates curl software-properties-common -y
+              curl -fsSL https://download.docker.com/linux/ubuntu/gpg | sudo apt-key add -
+              sudo add-apt-repository "deb [arch=arm64] https://download.docker.com/linux/ubuntu `lsb_release -cs` stable"
+              sudo apt update
+              sudo apt-get install -y docker-ce docker-ce-cli containerd.io docker-compose-plugin
+              sudo systemctl restart docker
+            displayName: Install moby
```

Comment:
I believe overall Microsoft is trying to discourage directly downloading from the 3rd party repo. Now that Mariner OS becoming more mature, I wonder if there is an equivalent of these packages store in PMC. If so, we should use them here.

Reply:
the arm64 machine is running ubuntu since mariner is not offered on that platform yet. i checked the PMC offering from ubuntu on this specific version and there was not a docker version that would work for our case. not tied to this change but was not able to find a better way when i was adding it.

Reply:
We need to install docker for ubuntu and not mariner. the underlying machine for this build process is still using Ubuntu since Mariner will not be added there for a few months.  When doing the initial changes for this i attempted to add the correct PMC link and call  "sudo apt-get install moby-engine" but no aarch64 packages were found.